docs/userguide/searchspace.md

```diff
@@ -1,46 +1,247 @@
+This method tries to infer the parameter types during the creation of the subspace.
+The parameter type is only necessary if special parameters should be used.
+
+```python
+df = pd.DataFrame(data=[[2, 5, 9], [3, 4, 7]], columns=["x0", "x1", "x2"])
+subspace = SubspaceDiscrete.from_dataframe(df)
+```
+
+In this case, it deduces that the parameters should be [`NumericalDiscreteParameter`](baybe.parameters.numerical.NumericalDiscreteParameter) and that no special encoding should be used.
+
+~~~
+ Discrete Parameters
+   Name                        Type  Num_Values Encoding
+ 0   x0  NumericalDiscreteParameter           2     None
+ 1   x1  NumericalDiscreteParameter           2     None
+ 2   x2  NumericalDiscreteParameter           2     None
+~~~
+
+### Creating a Simplex-Bound `SubspaceDiscrete`
+
+[`SubspaceDiscrete.from_simplex`](baybe.searchspace.discrete.SubspaceDiscrete.from_simplex) can be used to efficiently create a discrete search space (or discrete subspace) that is restricted by a simplex constraint, limiting the maximum sum of the parameters per dimension.
+This method uses a shortcut that removes invalid candidates already during the creation of parameter combinations and avoids to first create the full product space before filtering it.
+
+In the following example, a naive construction of the subspace would first construct the full product space, containing 25 points, although only 15 points are actually part of the simplex.
+
+```python
+parameters = [
+    NumericalDiscreteParameter(name="p1", values=[0, 0.25, 0.5, 0.75, 1]),
+    NumericalDiscreteParameter(name="p2", values=[0, 0.25, 0.5, 0.75, 1]),
+]
+subspace = SubspaceDiscrete.from_simplex(max_sum=1.0, simplex_parameters=parameters)
+```
+
+~~~
+       p1    p2
+ 0   0.00  0.00
+ 1   0.00  0.25
+ 2   0.00  0.50
+ ..   ...   ...
+ 12  0.75  0.00
+ 13  0.75  0.25
+ 14  1.00  0.00
+ 
+ [15 rows x 2 columns]
+~~~
+
+### Representation of data within discrete subspaces
+
+Internally, discrete subspaces are represented by two dataframes, the *experimental* and the *computational* representation.
+
+The experimental representation (`exp_rep`) contains all parameters as they were provided upon the construction of the search space and viewed by the experimenter. The computational representation (`comp_rep`) contains a representation of parameters that is actually used for the internal calculation.
+
+In particular, the computational representation contains no more labels or constant columns. This happens e.g. for [`SubstanceParameter`](baybe.parameters.substance.SubstanceParameter) or [`CategoricalParameter`](baybe.parameters.categorical.CategoricalParameter).
+
+The following example demonstrates the difference:
+```python
+from baybe.parameters import NumericalDiscreteParameter, CategoricalParameter
+from baybe.searchspace import SearchSpace
+
+speed = CategoricalParameter("Speed", values=["slow", "normal", "fast"], encoding="INT")
```

Comment:
this overall example is perfect, I wonder if we should use OHE encoding to demo that these two representations dont even necessarily have same shape?

Reply:
Nice idea, changed.